src/common/util/include/openvino/util/file_util.hpp

```diff
@@ -223,8 +214,8 @@ inline bool file_exists(const std::string& path) {
 }
 
 std::string get_file_ext(const std::string& path);
-std::string get_directory(const std::string& path);
-std::string path_join(const std::vector<std::string>& paths);
+ov::util::Path get_directory(const ov::util::Path& path);
+std::string path_join(const std::vector<ov::util::Path>& paths);
```

Comment:
```suggestion
Path path_join(const std::vector<ov::util::Path>& paths);
```
The functions which returns string should not be used or even removed

There is more need to have something like
```cpp
template<class C>
Path path_join(C&& paths);
```
Where `C` can be container of Paths or optionally of strings, wstring to convert them to paths.
The `initializer_list` can be used as container (rvalue) which can help when using it like (no to vector conversion):
```join_path({folder1, folder2, filename});```

Reply:
Done